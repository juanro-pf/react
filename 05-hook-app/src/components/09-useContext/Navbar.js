import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink activeClassName='active' exact to="/" className="nav-link" >Home</NavLink>
                    <NavLink activeClassName='active' exact to="/login" className="nav-link" >Login</NavLink>
                    <NavLink activeClassName='active' exact to="/about" className="nav-link" >About</NavLink>
                </div>
            </div>
        </nav>
    );
};
