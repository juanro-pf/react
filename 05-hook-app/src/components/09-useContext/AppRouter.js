import React from 'react';
import { Navbar } from './Navbar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <div>

                    <Navbar />

                    <div className='container'>
                        <Switch>
                            <Route exact path='/' component= { HomeScreen } />
                            <Route exact path='/login' component= { LoginScreen } />
                            <Route exact path='/about' component= { AboutScreen } />
                            {/* <Route component= { HomeScreen } /> */}
                            <Redirect to='./' />
                        </Switch>
                    </div>

                </div>
            </Router>
        </div>
    );
};
