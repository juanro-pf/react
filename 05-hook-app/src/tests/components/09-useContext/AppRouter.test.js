import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

import '@testing-library/jest-dom';

describe('Testing AppRouter', () => {

    const user= {
        name: 'Juan Ro',
        email: 'email'
    };

    const wrapper= mount(
        <UserContext.Provider value= {user}
        >
            <AppRouter />
        </UserContext.Provider>
    );
    
    test('Snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
    

});
