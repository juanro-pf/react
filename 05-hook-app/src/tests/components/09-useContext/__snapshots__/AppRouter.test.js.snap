// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AppRouter Snapshot 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </nav>
    <div
      className="container"
    >
      <div>
        <h1>
          HomeScreen
        </h1>
        <hr />
        <pre />
      </div>
    </div>
  </div>
</div>
`;
